custom-checks/pmd/src/main/java/org/openhab/tools/analysis/pmd/UseSLF4JLoggerRule.java

```diff
@@ -0,0 +1,125 @@
+/**
+ * Copyright (c) 2010-2018 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.tools.analysis.pmd;
+
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import net.sourceforge.pmd.lang.ast.Node;
+import net.sourceforge.pmd.lang.java.ast.ASTClassOrInterfaceType;
+import net.sourceforge.pmd.lang.java.ast.ASTReferenceType;
+import net.sourceforge.pmd.lang.java.ast.ASTType;
+import net.sourceforge.pmd.lang.java.ast.ASTVariableDeclarator;
+import net.sourceforge.pmd.lang.java.rule.AbstractJavaRule;
+
+/**
+ * Checks if a logger other than the one provided by slf4j is used
+ *
+ * @author Lyubomir Papazov - Initial contribution
+ *
+ */
+public class UseSLF4JLoggerRule extends AbstractJavaRule {
+
+    //There is a very similar check created by PMD that checks if only one logger is used:
+    //https://github.com/pmd/pmd/blob/pmd/5.8.x/pmd-java/src/main/java/net/sourceforge/pmd/lang/java/rule/logging/MoreThanOneLoggerRule.java
+
+    private static final String lOG_LITERAL = "Log";
+    private static final String LOGGER_LITERAL = "Logger";
+    private static final String CLASS_FOR_NAME_FAILED_WARNING = "Unable to create class object for that name";
+
+    private static final Class<?> LOG4J_LOGGER;
+
+    private static final Class<?> JAVA_LOGGER;
+
+    private static final Class<?> SLF4J_LOGGER;
+
+    private static final Class<?> COMMON_LOGGING_LOG;
+
+    private static final Class<?> LOGBACK_LOGGER;
+
+    private static Logger logger = Logger.getLogger(UseSLF4JLoggerRule.class.getName());
+
+    static {
+        Class<?> clazz;
+        try {
+            clazz = Class.forName("org.apache.log4j.Logger");
+        } catch (Throwable t) {
```

Comment:
Can we just catch the `ClassNotFoundException` and can you make method instead of duplicating the same block of code?